lib/beacon/live_admin/live/page_editor_live/new.ex

```diff
@@ -34,11 +34,15 @@ defmodule Beacon.LiveAdmin.PageEditorLive.New do
   end
 
   defp build_new_page(site, [layout | _] = _layouts) do
+    id = Ecto.UUID.generate()
```

Comment:
Instead of pre populating dummy data here that will be rendered in the form, what if we populate the data (only if it's empty) on the actual call to build the page ast? (the page ast is used on the visual editor only)

https://github.com/BeaconCMS/beacon_live_admin/blob/afcdb7ceaa046d72d6a30ca1f92dc4b33b5af050/lib/beacon/live_admin/live/page_editor_live/form_component.ex#L177

`page` is resolved from the changeset so if the required fields are present (id, title, path) then it should work, otherwise we could fill with dummy data to make the visual editor work. This way user won't see the dummy data on the form.

Wdyt?

Reply:
Okay, I've been trying to see what changes with your suggestion. I think populating just to make it render is fine, however populating at the LiveView level is also important because it not only helps rendering the visual editor but it also provided an appropriate state to the assigns so that the page can be drafted, saved and also published.

Reply:
@leandrocp Please take a look to the demo, drag and drop functionality has now been enabled and feature is fully  functional!

Reply:
> I think populating just to make it render is fine

We can go with that for this version 👍🏻